ICE eii: multiple annotations

<!--
[31mICE[0m: Rustc ./mul.rs '' 'thread 'rustc' ($TID) panicked at compiler/rustc_passes/src/eii.rs:119:13: 'multiple not supported right now'', 'thread 'rustc' ($TID) panicked at compiler/rustc_passes/src/eii.rs:119:13: 'multiple not supported right now''
File: /tmp/im/mul.rs
-->

snippet:
````rust
#[eii(eii1)]
fn a() {}

#[eii(eii1)]
fn main() {}

````

Version information
````
rustc 1.94.0-nightly (3f4dc1e02 2025-12-14)
binary: rustc
commit-hash: 3f4dc1e02d759aa3c3807d4efc1f7f6e293536a5
commit-date: 2025-12-14
host: x86_64-unknown-linux-gnu
release: 1.94.0-nightly
LLVM version: 21.1.5

````

Possibly related line of code:
https://github.com/rust-lang/rust/blob/3f4dc1e02d759aa3c3807d4efc1f7f6e293536a5/compiler/rustc_passes/src/eii.rs#L113-L125

Command:
`/home/matthias/.rustup/toolchains/master/bin/rustc `

<details><summary><strong>Program output</strong></summary>
<p>

```
error[E0658]: use of unstable library feature `extern_item_impls`
 --> /tmp/icemaker_global_tempdir.H7kAaxdihhDG/rustc_testrunner_tmpdir_reporting.mZz7YiFX9vOj/mvce.rs:1:3
  |
1 | #[eii(eii1)]
  |   ^^^
  |
  = note: see issue #125418 <https://github.com/rust-lang/rust/issues/125418> for more information
  = help: add `#![feature(extern_item_impls)]` to the crate attributes to enable
  = note: this compiler was built on 2025-12-14; consider upgrading it if it is out of date

error[E0658]: use of unstable library feature `extern_item_impls`
 --> /tmp/icemaker_global_tempdir.H7kAaxdihhDG/rustc_testrunner_tmpdir_reporting.mZz7YiFX9vOj/mvce.rs:4:3
  |
4 | #[eii(eii1)]
  |   ^^^
  |
  = note: see issue #125418 <https://github.com/rust-lang/rust/issues/125418> for more information
  = help: add `#![feature(extern_item_impls)]` to the crate attributes to enable
  = note: this compiler was built on 2025-12-14; consider upgrading it if it is out of date

error[E0428]: the name `eii1` is defined multiple times
 --> /tmp/icemaker_global_tempdir.H7kAaxdihhDG/rustc_testrunner_tmpdir_reporting.mZz7YiFX9vOj/mvce.rs:4:1
  |
1 | #[eii(eii1)]
  | ------------ previous definition of the macro `eii1` here
...
4 | #[eii(eii1)]
  | ^^^^^^^^^^^^ `eii1` redefined here
  |
  = note: `eii1` must be defined only once in the macro namespace of this module

error: the `main` function cannot be declared in an `extern` block
 --> /tmp/icemaker_global_tempdir.H7kAaxdihhDG/rustc_testrunner_tmpdir_reporting.mZz7YiFX9vOj/mvce.rs:5:1
  |
5 | fn main() {}
  | ^^^^^^^^^


thread 'rustc' (3711041) panicked at compiler/rustc_passes/src/eii.rs:119:13:
multiple not supported right now
stack backtrace:
   0:     0x7f3299790173 - <<std[e49fa185a119b89b]::sys::backtrace::BacktraceLock>::print::DisplayBacktrace as core[17865207963548b5]::fmt::Display>::fmt
   1:     0x7f3299e10648 - core[17865207963548b5]::fmt::write
   2:     0x7f32997a67c6 - <std[e49fa185a119b89b]::sys::stdio::unix::Stderr as std[e49fa185a119b89b]::io::Write>::write_fmt
   3:     0x7f3299766938 - std[e49fa185a119b89b]::panicking::default_hook::{closure#0}
   4:     0x7f3299784043 - std[e49fa185a119b89b]::panicking::default_hook
   5:     0x7f32987741d7 - std[e49fa185a119b89b]::panicking::update_hook::<alloc[756b9f94e2773738]::boxed::Box<rustc_driver_impl[e1c4671be5271ce8]::install_ice_hook::{closure#1}>>::{closure#0}
   6:     0x7f3299784322 - std[e49fa185a119b89b]::panicking::panic_with_hook
   7:     0x7f3299766a2a - std[e49fa185a119b89b]::panicking::panic_handler::{closure#0}
   8:     0x7f329975dab9 - std[e49fa185a119b89b]::sys::backtrace::__rust_end_short_backtrace::<std[e49fa185a119b89b]::panicking::panic_handler::{closure#0}, !>
   9:     0x7f32997685ad - __rustc[df3bb69314ad135f]::rust_begin_unwind
  10:     0x7f3296a6df0c - core[17865207963548b5]::panicking::panic_fmt
  11:     0x7f329a2d0c4c - rustc_passes[9766a5655bde9e6]::eii::check_externally_implementable_items
  12:     0x7f329a2d022f - rustc_query_impl[462c6a5584cea8d4]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[462c6a5584cea8d4]::query_impl::check_externally_implementable_items::dynamic_query::{closure#2}::{closure#0}, rustc_middle[7055997c00c0bb98]::query::erase::Erased<[u8; 0usize]>>
  13:     0x7f329af725a6 - rustc_query_system[5be0c069f6d5c539]::query::plumbing::try_execute_query::<rustc_query_impl[462c6a5584cea8d4]::DynamicConfig<rustc_query_system[5be0c069f6d5c539]::query::caches::SingleCache<rustc_middle[7055997c00c0bb98]::query::erase::Erased<[u8; 0usize]>>, false, false, false>, rustc_query_impl[462c6a5584cea8d4]::plumbing::QueryCtxt, false>
  14:     0x7f329af7238e - rustc_query_impl[462c6a5584cea8d4]::query_impl::check_externally_implementable_items::get_query_non_incr::__rust_end_short_backtrace
  15:     0x7f3299eec481 - rustc_interface[50af632e35c6bce3]::passes::analysis
  16:     0x7f3299eec2e5 - rustc_query_impl[462c6a5584cea8d4]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[462c6a5584cea8d4]::query_impl::analysis::dynamic_query::{closure#2}::{closure#0}, rustc_middle[7055997c00c0bb98]::query::erase::Erased<[u8; 0usize]>>
  17:     0x7f329af725a6 - rustc_query_system[5be0c069f6d5c539]::query::plumbing::try_execute_query::<rustc_query_impl[462c6a5584cea8d4]::DynamicConfig<rustc_query_system[5be0c069f6d5c539]::query::caches::SingleCache<rustc_middle[7055997c00c0bb98]::query::erase::Erased<[u8; 0usize]>>, false, false, false>, rustc_query_impl[462c6a5584cea8d4]::plumbing::QueryCtxt, false>
  18:     0x7f329af72176 - rustc_query_impl[462c6a5584cea8d4]::query_impl::analysis::get_query_non_incr::__rust_end_short_backtrace
  19:     0x7f329b12cdd4 - <rustc_interface[50af632e35c6bce3]::passes::create_and_enter_global_ctxt<core[17865207963548b5]::option::Option<rustc_interface[50af632e35c6bce3]::queries::Linker>, rustc_driver_impl[e1c4671be5271ce8]::run_compiler::{closure#0}::{closure#2}>::{closure#2} as core[17865207963548b5]::ops::function::FnOnce<(&rustc_session[307f8e7a4263bd4]::session::Session, rustc_middle[7055997c00c0bb98]::ty::context::CurrentGcx, alloc[756b9f94e2773738]::sync::Arc<rustc_data_structures[30161969bbcb553f]::jobserver::Proxy>, &std[e49fa185a119b89b]::sync::once_lock::OnceLock<rustc_middle[7055997c00c0bb98]::ty::context::GlobalCtxt>, &rustc_data_structures[30161969bbcb553f]::sync::worker_local::WorkerLocal<rustc_middle[7055997c00c0bb98]::arena::Arena>, &rustc_data_structures[30161969bbcb553f]::sync::worker_local::WorkerLocal<rustc_hir[5d00f225b65c0257]::Arena>, rustc_driver_impl[e1c4671be5271ce8]::run_compiler::{closure#0}::{closure#2})>>::call_once::{shim:vtable#0}
  20:     0x7f329afc19da - rustc_interface[50af632e35c6bce3]::interface::run_compiler::<(), rustc_driver_impl[e1c4671be5271ce8]::run_compiler::{closure#0}>::{closure#1}
  21:     0x7f329af31d8e - std[e49fa185a119b89b]::sys::backtrace::__rust_begin_short_backtrace::<rustc_interface[50af632e35c6bce3]::util::run_in_thread_with_globals<rustc_interface[50af632e35c6bce3]::util::run_in_thread_pool_with_globals<rustc_interface[50af632e35c6bce3]::interface::run_compiler<(), rustc_driver_impl[e1c4671be5271ce8]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>
  22:     0x7f329af31b60 - <std[e49fa185a119b89b]::thread::lifecycle::spawn_unchecked<rustc_interface[50af632e35c6bce3]::util::run_in_thread_with_globals<rustc_interface[50af632e35c6bce3]::util::run_in_thread_pool_with_globals<rustc_interface[50af632e35c6bce3]::interface::run_compiler<(), rustc_driver_impl[e1c4671be5271ce8]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1} as core[17865207963548b5]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  23:     0x7f329af374f8 - <std[e49fa185a119b89b]::sys::thread::unix::Thread>::new::thread_start
  24:     0x7f32948969cb - <unknown>
  25:     0x7f329491aa0c - <unknown>
  26:                0x0 - <unknown>

error: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: please make sure that you have updated to the latest nightly

note: rustc 1.94.0-nightly (3f4dc1e02 2025-12-14) running on x86_64-unknown-linux-gnu

query stack during panic:
#0 [check_externally_implementable_items] check externally implementable items
#1 [analysis] running analysis passes on crate `mvce`
end of query stack
error: aborting due to 4 previous errors

Some errors have detailed explanations: E0428, E0658.
For more information about an error, try `rustc --explain E0428`.

```

</p>
</details>

<!--
query stack:
#0 [check_externally_implementable_items] check externally implementable items
#1 [analysis] running analysis passes on crate `mvce`

-->

--- a/compiler/rustc_passes/src/eii.rs
+++ b/compiler/rustc_passes/src/eii.rs
@@ -116,7 +116,8 @@ pub(crate) fn check_externally_implementable_items<'tcx>(tcx: TyCtxt<'tcx>, (): 
         }
 
         if default_impls.len() > 1 {
-            panic!("multiple not supported right now");
+            let decl_span = tcx.def_ident_span(decl_did).unwrap();
+            tcx.dcx().span_delayed_bug(decl_span, "multiple not supported right now");
         }
 
         let (local_impl, is_default) =

--- a/tests/ui/eii/default/multiple-default-impls-same-name.rs
+++ b/tests/ui/eii/default/multiple-default-impls-same-name.rs
@@ -1,0 +1,19 @@
+#![crate_type = "lib"]
+#![feature(extern_item_impls)]
+// `eii` expands to, among other things, `macro eii() {}`.
+// If we have two eiis named the same thing, we have a duplicate definition
+// for that macro. The compiler happily continues compiling on duplicate
+// definitions though, to emit as many diagnostics as possible.
+// However, in the case of eiis, this can break the assumption that every
+// eii has only one default implementation, since the default for both eiis will
+// name resolve to the same eii definiton (since the other definition was duplicate)
+// This test tests for the previously-ICE that occurred when this assumption
+// (of 1 default) was broken which was reported in #149982.
+
+#[eii(eii1)]
+fn a() {}
+
+#[eii(eii1)]
+//~^ ERROR the name `eii1` is defined multiple times
+fn a() {}
+//~^ ERROR the name `a` is defined multiple times

--- a/tests/ui/eii/default/multiple-default-impls-same-name.stderr
+++ b/tests/ui/eii/default/multiple-default-impls-same-name.stderr
@@ -1,0 +1,25 @@
+error[E0428]: the name `a` is defined multiple times
+  --> $DIR/multiple-default-impls-same-name.rs:18:1
+   |
+LL | fn a() {}
+   | ------ previous definition of the value `a` here
+...
+LL | fn a() {}
+   | ^^^^^^ `a` redefined here
+   |
+   = note: `a` must be defined only once in the value namespace of this module
+
+error[E0428]: the name `eii1` is defined multiple times
+  --> $DIR/multiple-default-impls-same-name.rs:16:1
+   |
+LL | #[eii(eii1)]
+   | ------------ previous definition of the macro `eii1` here
+...
+LL | #[eii(eii1)]
+   | ^^^^^^^^^^^^ `eii1` redefined here
+   |
+   = note: `eii1` must be defined only once in the macro namespace of this module
+
+error: aborting due to 2 previous errors
+
+For more information about this error, try `rustc --explain E0428`.

--- a/tests/ui/eii/default/multiple-default-impls.rs
+++ b/tests/ui/eii/default/multiple-default-impls.rs
@@ -1,0 +1,8 @@
+#![crate_type = "lib"]
+#![feature(extern_item_impls)]
+
+#[eii(eii1)]
+fn a() {}
+
+#[eii(eii1)]
+fn b() {}

--- a/tests/ui/eii/default/multiple-default-impls.rs
+++ b/tests/ui/eii/default/multiple-default-impls.rs
@@ -1,8 +1,18 @@
 #![crate_type = "lib"]
 #![feature(extern_item_impls)]
+// `eii` expands to, among other things, `macro eii() {}`.
+// If we have two eiis named the same thing, we have a duplicate definition
+// for that macro. The compiler happily continues compiling on duplicate
+// definitions though, to emit as many diagnostics as possible.
+// However, in the case of eiis, this can break the assumption that every
+// eii has only one default implementation, since the default for both eiis will
+// name resolve to the same eii definiton (since the other definition was duplicate)
+// This test tests for the previously-ICE that occurred when this assumption
+// (of 1 default) was broken which was reported in #149982.
 
 #[eii(eii1)]
 fn a() {}
 
 #[eii(eii1)]
+//~^ ERROR the name `eii1` is defined multiple times
 fn b() {}

--- a/tests/ui/eii/default/multiple-default-impls.stderr
+++ b/tests/ui/eii/default/multiple-default-impls.stderr
@@ -1,0 +1,14 @@
+error[E0428]: the name `eii1` is defined multiple times
+  --> $DIR/multiple-default-impls.rs:16:1
+   |
+LL | #[eii(eii1)]
+   | ------------ previous definition of the macro `eii1` here
+...
+LL | #[eii(eii1)]
+   | ^^^^^^^^^^^^ `eii1` redefined here
+   |
+   = note: `eii1` must be defined only once in the macro namespace of this module
+
+error: aborting due to 1 previous error
+
+For more information about this error, try `rustc --explain E0428`.

